### PR TITLE
build: fixes rpm build when using defaults.

### LINKED ISSUE #12980: build rpm, get "*/tmp/bin/*" not exist
version: 5.2.2
OS: fedore-28

I clone the latest version and do chinesization. Then I want to build a rpm. But I got a error.
Here is what I run before:
```bash
go run build.go setup
go run build.go build
go run build.go pkg-rpm
```
Here is the main error (there are also some output about frontend, but it seem that not error):
```
mkdir -p /tmp/grafana-linux-pack382846784/usr/share/grafana
mkdir -p /tmp/grafana-linux-pack382846784/etc/grafana
mkdir -p /tmp/grafana-linux-pack382846784/etc/init.d
mkdir -p /tmp/grafana-linux-pack382846784/etc/sysconfig
mkdir -p /tmp/grafana-linux-pack382846784/usr/lib/systemd/system
mkdir -p /tmp/grafana-linux-pack382846784/usr/sbin
cp -p /home/code/go/src/github.com/grafana/grafana/tmp/bin/grafana-server /tmp/grafana-linux-pack382846784/usr/sbin/grafana-server
cp: cannot stat '/home/code/go/src/github.com/grafana/grafana/tmp/bin/grafana-server': No such file or directory
exit status 1
exit status 1
```

I try to find about how to build the grafana rpm, but I get not thing.
So I read the code to guest how to build the rpm.
Could you tell me how to build the rpm package of grafana?
Thanks!

### PR DESCRIPTION
**Pull request by xlson**
Fixes #12980

rpm and deb builds were failing when pkg-arch wasn't specified. This sets defaults.

### CODE REVIEW

File: build.go [hunk 1]
```diff
@@ -133,22 +144,6 @@ func main() {
 	}
 }
 
-func packageGrafana() {
```
Comment (marefr) [inline build.go]: Unsure what this actually did do and why you removed it :smile:

Comment (torkelo) [reply build.go]: Looks like it set platformArg and pkArch then called createLinuxPackages, pkgArch is now set on line 68 if not defined, and platform arg is now added on line 402 it seems.

Comment (xlson) [reply build.go]: Yeah, I moved most of the logic out of the function to be able to support sensible defaults. That resulted in the function being too small to be needed.

**Comment by marefr** [comment]
LGTM then

**Pull request merged**